odl/tomo/geometry/conebeam.py

```diff
@@ -111,24 +89,96 @@ def motion_grid(self):
-        return self._motion_grid
+    @abstractmethod
+    def src_position(self, mpar):
+        """The source position function.
 
-    @property
-    def src_radius(self):
-        """Source circle radius of this geometry."""
-        return self._src_radius
+        Parameters
+        ----------
+        mpar : element of motion parameters `motion_params`
+            Motion parameter for which to calculate the source position
 
-    @property
-    def det_radius(self):
-        """Detector circle radius of this geometry."""
-        return self._det_radius
+        Returns
+        -------
+        pos : `numpy.ndarray`, shape (`ndim`,)
+            The source position, a `ndim`-dimensional vector
+        """
+        raise NotImplementedError
+
+    def det_to_src(self, mpar, dpar, normalized=True):
+        """Vector pointing from a detector location to the source.
+
+        A function of the motion and detector parameters.
+
+        The default implementation uses the `det_point_position` and
+        `src_position` functions. Implementations can override this, for
+        example if no source position is given.
+
+        Parameters
+        ----------
+        mpar : element of motion parameters `motion_params`
+            Motion parameter at which to evaluate
+        dpar : element of detector parameters `det_params`
+            Detector parameter at which to evaluate
+        normalized : `bool`, optional
+            If `True`, return a normalized (unit) vector. Default: `True`
+
+        Returns
+        -------
+        vec : `numpy.ndarray`, shape (`ndim`,)
+            (Unit) vector pointing from the detector to the source
+        """
+
+        if mpar not in self.motion_params:
+            raise ValueError('mpar {} is not in the valid range {}.'
+                             ''.format(mpar, self.motion_params))
+        if dpar not in self.det_params:
+            raise ValueError('detector parameter {} is not in the valid '
+                             'range {}.'.format(dpar, self.det_params))
+
+        vec = self.det_point_position(mpar, dpar) - self.src_position(mpar)
+
+        if normalized:
+            vec /= np.linalg.norm(vec, axis=-1)
+
+        return vec
+
+    def __repr__(self):
+        """Return ``repr(self)``"""
+
+        # TODO
+
+        inner_fstr = '{!r}, {!r}, src_radius={}, det_radius={}'
+        if self.has_motion_sampling:
+            inner_fstr += ',\n agrid={agrid!r}'
+        if self.has_det_sampling:
+            inner_fstr += ',\n dgrid={dgrid!r}'
+        inner_str = inner_fstr.format(self.motion_params, self.det_params,
+                                      self.src_radius, self.det_radius,
+                                      agrid=self.motion_grid,
+                                      dgrid=self.det_grid)
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+
+class AxisOrientedGeometry(object):
+
```

Comment:
Should this class moved to `geometry` since it will be used also for the parallel 3D case?